### PR TITLE
Latency Performance Tests

### PR DESCRIPTION
**Pull request by prudmiry**
Latency Performance Tests

ib_send_lat,ib_write_lat,ib_read_lat,ib_atomic_lat

Signed-off-by: prudhvi <mprudhvi@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by prudmiry** [comment]
[job.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/510630/job.txt)
log added

File: io/net/infiniband/ib_latency_perf.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        cmd = "ssh %s %s" % (self.PEER_IP, msg)
+        if process.system(cmd, shell=True, ignore_status=True) != 0:
+            self.fail("test failed")
+
+    def test_ib_latency(self):
+        '''
+        test options are mandatory
+        ext test options are depends upon user
+        '''
+        tool_name = self.params.get("tool")
```
Comment (narasimhan-v) [inline io/net/infiniband/ib_latency_perf.py]: can be moved to `setUp()`

File: io/net/infiniband/ib_latency_perf.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        self.log.info("test with %s" % (tool_name))
+        test_op = self.params.get("test_opt", default="").split(",")
+        for val in test_op:
+            self.latencyperf_exec(tool_name, val, "")
+        ext_test_op = self.params.get("ext_opt", default="").split(",")
```
Comment (narasimhan-v) [inline io/net/infiniband/ib_latency_perf.py]: can be moved to `setUp()`

File: io/net/infiniband/ib_latency_perf.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        interfaces = netifaces.interfaces()
+        self.flag = self.params.get("ext_flag", default="0")
+        self.IF = self.params.get("Iface", default="")
+        self.PEER_IP = self.params.get("PEERIP", default="")
+        if self.IF not in interfaces:
+            self.skip("%s interface is not available" % self.IF)
+        if self.PEER_IP == "":
+            self.skip("%s peer machine is not available" % self.PEER_IP)
+        self.CA = self.params.get("CA_NAME", default="mlx4_0")
+        self.PORT = int(self.params.get("PORT_NUM", default="1"))
```
Comment (narasimhan-v) [inline io/net/infiniband/ib_latency_perf.py]: No specific use case to convert this value to int

File: io/net/infiniband/ib_latency_perf.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        '''
+        latency performance exec function
+        '''
+        logs = "> /tmp/ib_log 2>&1 &"
+        msg = "timeout %s %s -d %s -i %d %s %s %s" \
+            % (self.to, arg1, self.PEER_CA, self.PEER_PORT,
+                arg2, arg3, logs)
+        cmd = "ssh %s %s" % (self.PEER_IP, msg)
+        if process.system(cmd, shell=True, ignore_status=True) != 0:
+            self.fail("ssh failed to remote machine")
```
Comment (narasimhan-v) [inline io/net/infiniband/ib_latency_perf.py]: Does this mean ssh failure only ? or could it also be test command failure ?

File: io/net/infiniband/ib_latency_perf.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+        if process.system(tmp, shell=True, ignore_status=True) != 0:
+            self.fail("test failed")
+        self.log.info("server data for %s(%s)" % (arg1, arg2))
+        self.log.info("%s -d %s -i %d %s %s"
+                      % (arg1, self.PEER_CA, self.PEER_PORT,
+                         arg2, arg3))
+        msg = "timeout %s cat /tmp/ib_log; rm -rf /tmp/ib_log" % self.to
+        cmd = "ssh %s %s" % (self.PEER_IP, msg)
+        if process.system(cmd, shell=True, ignore_status=True) != 0:
+            self.fail("test failed")
```
Comment (narasimhan-v) [inline io/net/infiniband/ib_latency_perf.py]: Does this mean test failure only ? or could it also be ssh failure ?

File: io/net/infiniband/ib_latency_perf.py [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+            % (self.to, arg1, self.PEER_CA, self.PEER_PORT,
+                arg2, arg3, logs)
+        cmd = "ssh %s %s" % (self.PEER_IP, msg)
+        if process.system(cmd, shell=True, ignore_status=True) != 0:
+            self.fail("ssh failed to remote machine")
+        time.sleep(2)
+        self.log.info("client data for %s(%s)" % (arg1, arg2))
+        self.log.info("%s -d %s %s -i %d %s %s"
+                      % (arg1, self.CA, self.PEER_IP, self.PORT,
+                         arg2, arg3))
```
Comment (narasimhan-v) [inline io/net/infiniband/ib_latency_perf.py]: Is this really needed ? `process.system()` displays the command run

File: io/net/infiniband/ib_latency_perf.py.data/README.txt [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+-----------------------------
+Inputs Needed Run Tests:
+-----------------------------
+PEER_IP	    - IP of the Peer interface to be tested
+install nteifaces using pip
+  command: pip install netifaces
+install perf package
+Note:
+-----
+1. Generate sshkey for your test partner to run the test uninterrupted.
```
Comment (narasimhan-v) [inline io/net/infiniband/ib_latency_perf.py.data/README.txt]: Please improve on the readme file presentation.
Have sections like: Description, Input, Requirements.
Also, mention that MOFED is required.

File: io/net/infiniband/ib_latency_perf.py [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+
+    def test_ib_latency(self):
+        '''
+        test options are mandatory
+        ext test options are depends upon user
+        '''
+        tool_name = self.params.get("tool")
+        self.log.info("test with %s" % (tool_name))
+        test_op = self.params.get("test_opt", default="").split(",")
+        for val in test_op:
```
Comment (narasimhan-v) [inline io/net/infiniband/ib_latency_perf.py]: Multiplexer file can be effectively used in such cases, if you want to.

**Comment by prudmiry** [comment]
closed current PR and opening new PR--> IB Latency Performance Tests #96

**Pull request closed without merging**